javascripts/2048.js

```diff
@@ -1,32 +1,347 @@
+      // tile.attr("data-row","r0"); // unless a tile is there
+      var cols = makeCols(tile);
+      moveColUp(cols);
+      collideTilesUp(cols);
+      var reTile = $('.tile');
+      var reCol = makeCols(reTile);
+      moveColUp(reCol);
       break;
     case 40: //down
-      tile.attr("data-row","r3");
+      // tile.attr("data-row","r3");
+      var cols = makeCols(tile);
+      moveColDown(cols);
+      collideTilesDown(cols);
+      var reTile = $('.tile');
+      var reCol = makeCols(reTile);
+      moveColDown(reCol);
       break;
     case 37: //left
-      tile.attr("data-col","c0");
+      var rows = makeRows(tile);
+      moveRowLeft(rows);
+      collideTilesLeft(rows);
+      var reTile = $('.tile');
+      var reRow = makeRows(reTile);
+      moveRowLeft(reRow);
       break;
     case 39: //right
-      tile.attr("data-col","c3");
+      var rows = makeRows(tile);
+      moveRowRight(rows);
+      collideTilesRight(rows);
+      var reTile = $('.tile');
+      var reRow = makeRows(reTile);
+      moveRowRight(reRow);
       break;
   }
 }
+
+function makeRows(tile) {
+  var row0 = [];
+  var row1 = [];
+  var row2 = [];
+  var row3 = [];
+
+  for (var i = 0; i < tile.length; i++) {
+      if (tile[i].getAttribute("data-row") == "r0") {
+        row0.push(tile[i]);
+      } else if (tile[i].getAttribute("data-row") == "r1") {
+        row1.push(tile[i]);
+      } else if (tile[i].getAttribute("data-row") == "r2") {
+        row2.push(tile[i]);
+      } else if (tile[i].getAttribute("data-row") == "r3") {
+        row3.push(tile[i]);
+      }
+    }
+
+  var rows = [row0, row1, row2, row3];
+
+  return rows;
+}
+
+
+function collideTilesLeft(rows){
+
+  for (var i = 0; i < rows.length; i++) {
+    for (var j = 0; j< rows[i].length-1; j++){
+      var leftTileValue = rows[i][j].getAttribute("data-val");
+      var nextTileValue = rows[i][j+1].getAttribute("data-val");
+      if (leftTileValue == nextTileValue){
+        var newTileValue = leftTileValue * 2;
+        rows[i][j].setAttribute("data-val", newTileValue);
+        rows[i][j].textContent = newTileValue;
+        rows[i][j+1].remove();
+    };
+
+  }
+}
+}
+
+
+function moveRowLeft(row) {
+  // for(var i = 0; i < row.length; i++) {
+  //   row[i].setAttribute("data-col", ("c" + i));
+  //   }
+  for (var i = 0; i < row.length; i++) {
+  for(var j = 0; j < row[i].length; j++) {
+    row[i][j].setAttribute("data-col", ("c" + j));
+    }
+  }
+}
+
+
+function collideTilesRight(rows){
+  for (var i = 0; i < rows.length; i++) {
+
+  for (var j = 0; j < rows[i].length -1; j++){
+   var leftTileValue = rows[i][j].getAttribute("data-val");
+   var nextTileValue = rows[i][j+1].getAttribute("data-val");
+      if (leftTileValue == nextTileValue){
+        var newTileValue = nextTileValue * 2;
+          rows[i][j+1].setAttribute("data-val", newTileValue);
+          rows[i][j+1].textContent = newTileValue;
+          rows[i][j].remove();
+    };
+
+  }
+}
+}
```

Comment:
Oof, be careful. Having blocks end like this can make your code difficult to read